vertx-web/src/main/java/io/vertx/ext/web/sstore/impl/SessionImpl.java

```diff
@@ -74,6 +75,18 @@ public String id() {
   }
 
   @Override
+  public Session regenerateId() {
+    if (oldId == null) {
+      // keep track of just the first one since the
+      // regeneration during the remaining lifecycle are ephemeral
+      oldId = id;
+    }
+    id = UUID.randomUUID().toString();
```

Comment:
@garricko thanks for spotting that, I've updated the code to respect the min length configuration. Regarding the possible blocking I'm not sure if I can do anything about it. If someone expects to have thousands of sessions I'd expect that that user would invest some money on a TRNG device or use some "modern" CPU like the Intel Ivy Bridge which has an integrated HWRNG circuit. If someone is on a budget, they must accept the limitations and can work around the speed by redirecting the secure random source data to `/dev/urandom`